sklearn/metrics/_ranking.py

```diff
@@ -362,12 +372,69 @@ def roc_auc_score(y_true, y_score, *, average="macro", sample_weight=None,
 
     Examples
     --------
-    >>> import numpy as np
+    ROC-AUC can be computed in different problem. In binary case, both
+    probability estimates and non-thresholded decision values can be used:
```

Comment:
I feel like the docstring examples should be strictly minimal, just to briefly illustrate the API. Anything longer than a few lines with some explanations should probably be in the UG?

Reply:
OK so we could remove the printing indeed. But I think that we should highlight the 3 usages cases (binary, multilabel and multiclass). And I can add the narrative in the UG.

@NicolasHug Do you have any idea how to make `roc_auc_score` to work with multilabel :)